### PR TITLE
add POST /api/articles/:article_id/comments endpoint to add comments …

### PR DESCRIPTION
**Pull request by ProdiGeeZ**
…to articles, testing for errors and correct functionality

### CODE REVIEW

File: __tests__/app.tests.js [hunk 1]
```diff
@@ -159,26 +159,91 @@ describe('GET /api/articles/:article_id/comments', () => {
...
-                expect(response.body.msg).toBe("Bad Request: Invalid article_id format")
+                expect(response.body.msg).toBe("Bad Request: Invalid article_id format.")
             });
     });
-});
\ No newline at end of file
+});
+
+describe('POST /api/articles/:article_id/comments', () => {
+    test('201: Should create and post a comment with 2 keys from the request body being username and body, including the other 3 keys', () => {
```
Comment (nickkagame) [inline __tests__/app.tests.js:187]: could have another 201 that allows a comment with extra properties to go through (the extra properties would be ignored)

File: __tests__/app.tests.js [hunk 2]
```diff
@@ -159,26 +159,91 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+        });
+});
+test('400: Should return an error message when the article_id input is invalid ', () => {
+    const commentData = { username: "butter_bridge", body: "first comment lol." };
+    return request(app)
+        .post("/api/articles/bad_id/comments")
+        .send(commentData)
+        .expect(400)
+        .then((response) => {
+            expect(response.body.msg).toBe("Bad Request: Invalid article_id format.")
```
Comment (nickkagame) [inline __tests__/app.tests.js:247]: good testing

File: controllers/comments.controller.js [hunk 3]
```diff
@@ -1,10 +1,27 @@
...
         })
         .catch(next);
-};
\ No newline at end of file
+};
+
+exports.postComment = (req, res, next) => {
+    const { article_id } = req.params;
+    const commentData = req.body;
+    return articleCheck(article_id)
```
Comment (nickkagame) [inline controllers/comments.controller.js:19]: this isn't really necessary as article_id is a foreign key on the comments table and therefore will have to exist.

File: models/comments.model.js [hunk 4]
```diff
@@ -28,5 +29,30 @@ exports.fetchArticleComments = (article_id) => {
...
+        commentData.body,
+        article_id,
+        commentData.username,
+        0,
+        new Date(),
+    );
+    return db.query(queryString)
+        .then((result) => {
+            if (result.rows.length === 0) {
+                console.log('Hey buddy.. you fkd up');
```
Comment (nickkagame) [inline models/comments.model.js:44]: You can do a Promise.reject here, if you remove the check exists, as empty rows will mean article doesn't exist.

**Review by nickkagame — APPROVED** [review]
Nicely done mohammed! Just a few comments that you might want to act on.

File: models/comments.model.js (hunk 4 above)
Comment (ProdiGeeZ) [reply models/comments.model.js:44]: I forgot I left that in here 😭 sorry nick ahaha

**Pull request merged**